frame/multi-account/src/lib.rs

```diff
@@ -0,0 +1,1506 @@
+for SigsLen<AccountId>
+{
+	fn weigh_data(&self, (_, sigs): (&u16, &Vec<AccountId>)) -> Weight {
+		10_000 * (sigs.len() as u32 + 1)
+	}
+}
+impl<AccountId> ClassifyDispatch<(&u16, &Vec<AccountId>)>
+for SigsLen<AccountId>
+{
+	fn classify_dispatch(&self, _: (&u16, &Vec<AccountId>))
+		-> DispatchClass
+	{
+		DispatchClass::Normal
+	}
+}
+impl<AccountId> PaysFee<(&u16, &Vec<AccountId>)>
+for SigsLen<AccountId>
+{
+	fn pays_fee(&self, _: (&u16, &Vec<AccountId>)) -> bool {
+		true
+	}
+}
+
+/// A module identifier. These are per module and should be stored in a registry somewhere.
+#[derive(Clone, Copy, Eq, PartialEq, Encode, Decode)]
+struct IndexedMultiAccountModuleId(u16);
+
+impl TypeId for IndexedMultiAccountModuleId {
+	const TYPE_ID: [u8; 4] = *b"muac";
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		type Error = Error<T>;
+
+		/// Deposit one of this module's events by using the default implementation.
+		fn deposit_event() = default;
+
+		/// Create a new multi account with a threshold and given signatories.
+		///
+		/// This generates a new static account id, that will persist when the threshold or set of signatories are
+		/// changed later on.
+		///
+		/// Payment: `MultiAccountDepositBase` will be reserved, plus the number of signatories times
+		/// `MultiAccountDepositFactor`. It is returned when the multi account is updated or removed.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// - `threshold`: The total number of approvals dispatches from this multi account require before they are
+		///    executed.
+		/// - `other_signatories`: The accounts (other than the sender) who can approve dispatches from this multi
+		///    account. May not be empty.
+		///
+		/// # <weight>
+		/// - `O(S)`.
+		/// - One balance-reserve operation.
+		/// - One insert with `O(S)` where `S` is the number of
+		///   signatories. `S` is capped by `MaxSignatories`, with weight being proportional.
+		/// - One encode & hash, both of complexity `O(S)`.
+		/// - I/O: 1 mutate `O(S)`
+		/// - One event.
+		/// - Storage: inserts one item, value size bounded by `MaxSignatories`, with a
+		///   deposit taken for its lifetime of
+		///   `MultiAccountDepositBase + other_signatories.len() * MultiAccountDepositFactor`.
+		/// # </weight>
+		#[weight = <SigsLen<T::AccountId>>::new()]
+		fn create(origin,
+			threshold: u16,
+			other_signatories: Vec<T::AccountId>
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			ensure!(threshold >= 1, Error::<T>::ZeroThreshold);
+			let max_sigs = T::MaxSignatories::get() as usize;
+			ensure!(!other_signatories.is_empty(), Error::<T>::TooFewSignatories);
+			ensure!(threshold as usize <= other_signatories.len() + 1, Error::<T>::TooFewSignatories);
+			ensure!(other_signatories.len() < max_sigs, Error::<T>::TooManySignatories);
+			ensure!(other_signatories.len() < u32::max_value() as usize, Error::<T>::TooManySignatories);
```

Comment:
This check is useless?

The line above already confirms this is true.

Reply:
Good point, removed